verifier/source_reader_service.go

```diff
@@ -33,28 +32,28 @@ const (
 
 // SourceReaderService wraps a SourceReader and converts MessageSentEvents to VerificationTasks.
 type SourceReaderService struct {
-	services.StateMachine
+	sync   services.StateMachine
+	stopCh services.StopChan
 	wg     sync.WaitGroup
-	stopCh chan struct{}
 
-	sourceReader         chainaccess.SourceReader
-	headTracker          chainaccess.HeadTracker
 	logger               logger.Logger
-	lastProcessedBlock   *big.Int
+	sourceReader         chainaccess.SourceReader
 	verificationTaskCh   chan batcher.BatchResult[VerificationTask]
 	ccipMessageSentTopic string
 	pollInterval         time.Duration
 	chainSelector        protocol.ChainSelector
-	mu                   sync.RWMutex
 
+	// State that requires synchronization
+	mu                 sync.RWMutex
+	lastProcessedBlock *big.Int
 	// Reset coordination using optimistic locking pattern.
 	// This version counter is incremented each time ResetToBlock() is called.
 	// The processEventCycle() captures the version at the start of its cycle,
 	// and checks it again before updating lastProcessedBlock. If the version
 	// changed (indicating a reset occurred during the cycle's RPC calls),
 	// the cycle skips its update to avoid overwriting the reset value.
 	// This allows us to protect lastProcessedBlock without holding locks across I/O.
-	resetVersion atomic.Uint64
+	resetVersion uint64
```

Comment:
It was almost always accessed when mutex was acquired so making it `atomic.Uint64` is a bit misleading as we always rely on external locking